test/functional/editor/K_spec.lua

```diff
@@ -33,6 +33,10 @@ describe('K', function()
     feed('i'..test_file..'<ESC>K')
     retry(nil, nil, function() eq(1, eval('filereadable("'..test_file..'")')) end)
     eq({'fnord'}, eval("readfile('"..test_file.."')"))
+    -- Confirm that Neovim is still in terminal mode after K is pressed (#16692).
+    eq('t', eval('mode()'))
+    feed('<space>')  -- Any key, not just <space>, can be used here to escape.
+    eq('n', eval('mode()'))
```

Comment:
It seems when the file is written, the process may be still running (especially on Windows), so pressing `<Space>` doesn't always return to Normal mode. Something like a call to `helpers.sleep()` should be needed before these lines.

Reply:
I tried unsuccessfully using sleep. I'll try again with a higher value.

Reply:
My earlier failed attempt was with 100ms. Per your recommendation below, I've changed it to 500ms.